pkg/domain/infra/abi/images.go

```diff
@@ -858,14 +858,15 @@ func execSu(execUser *user.User, command []string) error {
 	return cmd.Run()
 }
 
-func execMachine(execUser *user.User, command []string, machinectl string) error {
-	verb := machinectl
-	args := []string{"shell", "-q", execUser.Username + "@.host"}
-	if execUser.Uid == "0" {
-		args = append([]string{verb}, args...)
-		verb = "sudo"
+func execRunUser(execUser *user.User, command []string, runUser string) error {
+	cmdLogin, err := utils.LoginUser(execUser.Username)
+	if err != nil {
+		return err
 	}
-	cmd := utils.CreateSCPCommand(exec.Command(verb, args...), command)
-	logrus.Debugf("Executing via machinectl: %q", cmd)
+	defer cmdLogin.Process.Kill()
```

Comment:
I think after the `Kill()` you either need to `Wait()` for the process or `Release()` it, otherwise you leak it